### ISSUE TITLE
Permissions settings are ignored by apps and not persisted

### ISSUE THREAD

**Issue by arkraft** [elementary/switchboard-plug-applications#171]
### What Happened?

It seems the permissions tab does not display the correct permissions for the apps or the apps do not follow the given permissions.
I removed the permission for "GNOME Web" to access network, just to try it out. After several restarts of the app i can still access the internet though. I expected "GNOME Web" to throw an error of any kind.
![grafik](https://user-images.githubusercontent.com/7426376/139454017-2d453993-e488-4f75-aa0d-ec4aa6726ad1.png)

Same with Screenshot, where i removed the permission to access system folders and was still able to save a screenshot inside the system folder.
![grafik](https://user-images.githubusercontent.com/7426376/139454353-3104f09f-fd59-4b31-8f15-87f57452bed9.png)
 
Or Evince was able to read all directories even though i removed access to them.
Maybe i just don't understand the permissions model of flatpak but this was like unexpected behavior for me.

Restarting the app or the system does not help.
After restarting switchboard and looking at the permissions again, they are not persisted and everything is back to the defaults. Adding permissions is persisted on the other hand, only the removed permissions are added again

### Steps to Reproduce

1. Go to the permissions tab
2. Select "GNOME Web"
3. Remove the permission to access networks
4. Start GNOME Web
5. Access any internet site
6. Close Switchboard
7. Open Switchboard again
8. Go to the permissions tab
9. Select GNOME Web

### Expected Behavior

It should not be able to access the internet. GNOME Web was expected to fail, just like if you are not connected to the internet.
After restarting switchboard i expected my settings to be persisted. All removed permissions are set to the default values. Only additional permissions are persisted

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

**Comment by leolost2605** [comment]
I can confirm this here with filesystem and device permissions as well. To double check I used flatseal. When toggling a permission in elementary settings, flatseal showed a little icon with "manually changed by the user" but the actual switch doesn't change

**Comment by myhousece** [comment]
Yes, I'm facing this same problem with permissions. They don't comply. I disable them and they come back. I thought it would be just on my system, but I don't think so! I'm experimenting with Elementary 8.0 and in this case, the same thing reported is happening.
![acesso app](https://github.com/user-attachments/assets/bb75bc3d-2286-4aa4-bb4f-4cabe072f11a)
